geodata/models/common.py

```diff
@@ -0,0 +1,73 @@
+import tempfile
+
+from django.conf import settings
+from django.contrib.gis.db import models
+from django.utils import timezone
+
+
+class DeferredFieldsManager(models.Manager):
+    def __init__(self, *deferred_fields):
+        self.deferred_fields = deferred_fields
+        super().__init__()
+
+    def get_queryset(self):
+        return super().get_queryset().defer(*self.deferred_fields)
+
+
+class ModifiableEntry(models.Model):
+    """A base class for models that need to track modified datetimes."""
+
+    modified = models.DateTimeField(editable=False, help_text='The last time this entry was saved.')
+    created = models.DateTimeField(editable=False, help_text='When this was added to the database.')
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
```

Comment:
In Girder, we track the modified date but also the user who did the modification.  Similarly, we track the user who created the object.  Do we want to do that here as well?  This would add a creator field and a modifier field.

Reply:
This is a good idea - I will add `creator` and `modifier` user fields.

Should we keep a history of the modifiers or is just updating the field with the laster user to modify it okay?

Reply:
I have to make `ModifiableEntry` non-abstract to do this.

Reply:
Also, access the user in the `save` method that I override isn't clear. I don't think the request is passed, so we may have to use a middleware plugin like this: https://blndxp.wordpress.com/2016/03/04/django-get-current-user-anywhere-in-your-code-using-a-middleware/

Reply:
I added these fields with null available. For now, the user isn't saved as I'm not sure what is the best way to access the current user

Reply:
I no longer set these variables in the `save()` event due to bad practice.